File: LeetCode/Easy/0504-base-7/0504-base-7.java
class Solution {
    public String convertToBase7(int num) {

        if(num == 0) return "0";

        StringBuilder sb = new StringBuilder();
        boolean minus = num < 0;
        num = Math.abs(num);
    
    
        while(num > 0) {
            sb.append(num % 7);
            num /= 7;
        }
        
        if(minus) {
            sb.append("-");
        }

        return sb.reverse().toString();
    }
}

File: LeetCode/Easy/0504-base-7/0504-base-7.kt
class Solution {
    fun convertToBase7(num: Int): String {
        if(num == 0) return "0";

        val sb: StringBuilder = StringBuilder()
        val minus: Boolean = num < 0
        var absNum = Math.abs(num)

        while(absNum > 0) {
            sb.append(absNum % 7)
            absNum /= 7 
        }

        if(minus) {
            sb.append("-")
        }
  
        return sb.reverse().toString()
    }
}